packages/lesmis/src/components/MainMenu.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ *
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import {
+  Typography,
+  IconButton,
+  Divider,
+  Drawer,
+  Link,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+} from '@material-ui/core';
+import { Link as RouterLink } from 'react-router-dom';
+import { Home, ImportContacts, ContactMail, Close, Menu } from '@material-ui/icons';
+import { LightIconButton } from '@tupaia/ui-components';
+import { FlexEnd } from './Layout';
+
+const StyledList = styled(List)`
+  width: 22.5rem;
+
+  .MuiListItem-gutters {
+    padding-left: 2rem;
+    padding-right: 2rem;
+  }
+
+  .MuiListItemIcon-root {
+    min-width: 2.5rem;
+  }
+
+  .MuiListItemText-primary {
+    font-size: 1.125rem;
+    line-height: 140%;
+  }
+`;
+
+const MenuSection = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+  margin-right: 1.2rem;
+  padding-right: 0.3rem;
+  margin-left: -1.3rem;
+  border-right: 1px solid rgba(0, 0, 0, 0.2);
+
+  .MuiIconButton-root {
+    padding: 1rem;
+  }
+`;
+
+const MenuTray = styled(FlexEnd)`
+  padding: 0.375rem 0.375rem 0 0;
+  margin-bottom: -0.375rem;
+`;
+
+const MenuHeading = styled(Typography)`
+  font-weight: normal;
+  font-size: 2rem;
+  line-height: 140%;
+  margin-bottom: 0.5rem;
+`;
+
+const StyledDivider = styled(Divider)`
+  margin: 0.375rem 1.875rem;
+`;
+
+const TupaiaText = styled(Typography)`
+  color: ${props => props.theme.palette.text.secondary};
+  margin-top: 0.5rem;
+  font-size: 0.75rem;
+  line-height: 140%;
+`;
+
+const ListItemLink = props => <ListItem button component={RouterLink} {...props} />;
+
+export const MainMenu = () => {
+  const [open, setOpen] = useState(false);
+
+  const toggleDrawer = isOpen => event => {
+    if (event.type === 'keydown' && (event.key === 'Tab' || event.key === 'Shift')) {
+      return;
+    }
+
+    setOpen(isOpen);
+  };
```

Comment:
So just to be clear, this means that if the menu drawer is expanded, and the user presses a key other than the `tab` or `shift` keys then it will close the menu?

I'm wondering if this is standard? Would perhaps just pressing the `esc` key to close it be a bit more common?

Reply:
hmmm now that I look at it in retrospect I'm not too sure why I did that. I actually stole it from the material-ui docs but I agree that it doesn't make too much sense. I have removed the event listener for keypresses and tested to check that 'enter', 'space' and 'esc' work as expected.